src/cmds/qstat.c

```diff
@@ -302,49 +359,174 @@ states(char *string, char *q, char *r, char *h, char *w, char *t, char *e, int l
  * @param[in] n - attribute name
  * @param[in] r - resource name
  * @param[in] v - value for the attribute
+ * @param[in] one_line - one line per attribute
  *
  * @return Void
  *
  */
 void
-prt_attr(char *n, char *r, char *v)
+prt_attr(char *n, char *r, char *v, int one_line)
 {
-	char *c;
-	char *comma = ",";
-	int   first = 1;
-	int   l;
-	int   start;
-
-	start = strlen(n) + 7;	/* 4 spaces + ' = ' is 7 */
-	printf("    %s", n);
-	if (r) {
-		start += strlen(r) + 1;
-		printf(".%s", r);
-	}
-	printf(" = ");
-	c = strtok(v, comma);
-	while (c) {
-		if ((l = strlen(c)) + start < 78) {
-			printf("%s", c);
-			start += l;
-		} else {
-			if (! first) {
-				printf("\n\t");
-				start = 9;
+	int            first = 1;
+	int            l = 0;
+	int            start = 0;
+	int            i = 0;
+	char           *c = NULL;
+	char           *comma = ",";
+	static char    *prev_resc_name = NULL;
+	char           *buf = NULL;
+	char           *keyvalpair = NULL;
+	char           *key = NULL;
+	char           *val = NULL;
+	char           *temp = NULL;
+
+	if(v == NULL)
+		return;
+	switch (output_format) {
+		case FORMAT_JSON:
+			if (strcmp(n, ATTR_v) == 0) {
+				if (add_json_node(JSON_OBJECT, JSON_NULL, n, NULL) == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+				keyvalpair = (char *) malloc(MAXPATHLEN);
+				l = MAXPATHLEN;
+				if(keyvalpair == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+				while(*v) {
+					/* value is splitted based on comma and stored in keyvalpair.
+					 * Escaped comma is treated as value and '\' is removed while parsing for json. */
+					while(!(*v == ',' || *v == '\0')) {
+						if(*v == '\\' && *(v+1) == ',')
+							v++;
+						keyvalpair[i++] = *v++;
+						if(i >= l-1) {
+							l *= BUFFER_GROWTH_RATE;
+							temp = (char *) realloc(keyvalpair, l);
+							if(temp == NULL) {
+								free(keyvalpair);
+								fprintf(stderr, "qstat: out of memory\n");
+								exit(1);
+							}
+							keyvalpair = temp;
+						}
+					}
+					keyvalpair[i] = '\0';
+					key = strtok_r(keyvalpair, "=", &val);
+					if(add_json_node(JSON_VALUE, JSON_NULL, key, val) == NULL) {
+						fprintf(stderr, "qstat: out of memory\n");
+						exit(1);
+					}
+					if(*v != '\0') {
+						i = 0; v++;
```

Comment:
to be more clear - it is not there in the guidelines that we can have two or more statements on the same line... Also, it is not mentioned in the guidelines that we cannot have such statements, and this is what i mentioned in my first comment that the standard is not clear on this, so get a clarification..  "usually, one line contains a single operation. Our standard does not cover this specific point. If you get me a confirmation that we can have two operations on one line, I will be ok. "
This style of code being used in other places doesn't guarantee that it is the right thing to do... I never said it is bad coding practice.. so I dont think we need a justification of any sort... The standard is not clear on this, so a clarification is needed...